web/src/main/webapp/tableTarget.ts

```diff
@@ -116,19 +116,25 @@ export class RemoteTableObject extends RemoteObject {
         });
     }
 
-    public createProjectToEigenVectorsRequest(r: RemoteObject, dimension: number): RpcRequest {
+    public createProjectToEigenVectorsRequest(r: RemoteObject, dimension: number, projectionName: string): RpcRequest {
         return this.createRpcRequest("projectToEigenVectors", {
             id: r.remoteObjectId,
-            numComponents: dimension
+            numComponents: dimension,
+            projectionName: projectionName
         });
     }
 
     public createFilterEqualityRequest(filter: EqualityFilterDescription): RpcRequest {
         return this.createRpcRequest("filterEquality", filter);
     }
 
-    public createCorrelationMatrixRequest(columnNames: string[]): RpcRequest {
-        return this.createRpcRequest("correlationMatrix", {columnNames: columnNames});
+    public createCorrelationMatrixRequest(columnNames: string[], totalRows: number, toSample: boolean): RpcRequest {
```

Comment:
There is a conflict here that you will have to resolve.
You will notice that I made these functions more type-safe; now RpcRequest has a type argument.